src/database/mod.rs

```diff
@@ -108,12 +134,179 @@ pub fn db_instance(db_path: &path::Path) -> Result<DbInstance, DatabaseError> {
     Ok(rows.pop().expect("No row in instances table?"))
 }
 
-// Sqlite supports up to i64, thus rusqlite prevents us from inserting u64's.
-// We use this to panic rather than inserting a truncated integer into the database (as we'd have
-// done by using `n as u32`).
-fn timestamp_to_u32(n: u64) -> u32 {
-    n.try_into()
-        .expect("Is this the year 2106 yet? Misconfigured system clock.")
+/// Register a new vault to be watched. Atomically inserts the vault and the Emergency signatures.
```

Comment:
Is it worth separating the vault registration and emergency signature storage, for the option of out-sourced WTs without emergency function in future?

Reply:
Definitely. Back in July when i wrote this code i had an answer to this, looking for traces about it now..

Reply:
So what i think is: it's awkward to have a single entity fill both Emergency and Cancel roles.

I *think* what i though then was:
- the protocol allows us to share only the sigs of either of the transaction (separated `sig` messages, re https://github.com/revault/practical-revault/issues/85) so it's possible to have a WT which doesn't expect Emer sigs
- "i'll correct this comment later"

Anyways i think the right approach to this would be to be able to start `miradord` either in "Emergency mode" or "Cancel mode" (or both?). It's because back when we designed the protocol we expected people to run their own WTs, now i believe most of them will want outsourced ones..

Reply:
yeah, having both is obviously better as it gives more options for deployments. Not sure how much more complexity it will add to the implementation... but seems do-able at a glance